skops/io/_trusted_types.py

```diff
@@ -10,6 +10,10 @@
 
 PRIMITIVE_TYPE_NAMES = ["builtins." + t.__name__ for t in PRIMITIVES_TYPES]
 
+BUILTIN_TYPES = [list, set, map, tuple]
```

Comment:
I wonder about the name. Above we have "primitive types" but they are also "builtin". Should this be "complex types", "container types", "composite types" or something along these lines?

Then we could have `BUILTIN_TYPE_NAMES = PRIMITIVE_TYPE_NAMES + COMPLEX_TYPE_NAMES` (or whatever name is chosen) and save a bit of code duplication.